src/System.Private.CoreLib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -2,218 +2,59 @@
-        {
-            // There may be a better way to do this.
-            // we got RuntimeMethodHandles here and we need to go to MethodBase
-            // but we don't know whether the reflection info has been initialized
-            // or not. So we call GetMethods and GetConstructors on the type
-            // and then we fetch the proper MethodBase!!
-            IntPtr mh = rgMethodHandle[i];
-
-            if (mh == IntPtr.Zero)
-                return null;
-
-            IRuntimeMethodInfo mhReal = RuntimeMethodHandle.GetTypicalMethodDefinition(new RuntimeMethodInfoStub(mh, this));
-
-            return RuntimeType.GetMethodBase(mhReal);
-        }
-
-        public virtual int GetOffset(int i) { return rgiOffset[i]; }
-        public virtual int GetILOffset(int i) { return rgiILOffset[i]; }
-        public virtual string GetFilename(int i) { return rgFilename == null ? null : rgFilename[i]; }
-        public virtual int GetLineNumber(int i) { return rgiLineNumber == null ? 0 : rgiLineNumber[i]; }
-        public virtual int GetColumnNumber(int i) { return rgiColumnNumber == null ? 0 : rgiColumnNumber[i]; }
-
-        public virtual bool IsLastFrameFromForeignExceptionStackTrace(int i)
-        {
-            return (rgiLastFrameFromForeignExceptionStackTrace == null) ? false : rgiLastFrameFromForeignExceptionStackTrace[i];
-        }
-
-        public virtual int GetNumberOfFrames() { return iFrameCount; }
-    }
-
-
-    // Class which represents a description of a stack trace
-    // There is no good reason for the methods of this class to be virtual.  
+    /// <summary>
+    /// Class which represents a description of a stack trace
+    /// There is no good reason for the methods of this class to be virtual.
+    /// In order to ensure trusted code can trust the data it gets from a
+    /// StackTrace, we use an InheritanceDemand to prevent partially-trusted
+    /// subclasses.
+    /// </summary>
     public class StackTrace
     {
-        private StackFrame[] frames;
         private int m_iNumOfFrames;
         public const int METHODS_TO_SKIP = 0;
         private int m_iMethodsToSkip;
-
-        // Constructs a stack trace from the current location.
+        
+        /// <summary>
+        /// Stack frames comprising this stack trace.
+        /// </summary>
+        private StackFrame[] _stackFrames;
+
+        /// <summary>
+        /// Constructs a stack trace from the current location.
+        /// </summary>
         public StackTrace()
         {
             m_iNumOfFrames = 0;
             m_iMethodsToSkip = 0;
             CaptureStackTrace(METHODS_TO_SKIP, false, null, null);
         }
 
-        // Constructs a stack trace from the current location.
-        //
-        public StackTrace(bool fNeedFileInfo)
+        /// <summary>
+        /// Constructs a stack trace from the current location.
+        /// </summary>
+        public StackTrace(bool needFileInfo)
```

Comment:
This should stay as `fNeedFileInfo`. The argument name is part of the public surface, and we cannot fix it, unfortunately.